core/src/main/java/org/easymock/internal/MocksControl.java

```diff
@@ -373,6 +373,24 @@ public void asStub() {
         }
     }
 
+    public IExpectationSetters<Object> atLeast(int minimalNumberOfTimes) {
+        try {
+            state.times(new Range(minimalNumberOfTimes, Integer.MAX_VALUE));
+            return this;
+        } catch (RuntimeExceptionWrapper e) {
+            throw (RuntimeException) e.getRuntimeException().fillInStackTrace();
+        }
+    }
+
+    public IExpectationSetters<Object> atMost(int maximalNumberOfTimes) {
+        try {
+            state.times(new Range(1, maximalNumberOfTimes));
```

Comment:
The minimum range is 0. However, the maximum can't be 0 for some historical
reason. So you can't have a `times(0, 0)` range (which would mean: "I want
that to be not called")

On 8 February 2016 at 12:15, Xavier Pigeon <notifications@github.com> wrote:

> In core/src/main/java/org/easymock/internal/MocksControl.java
> <https://github.com/easymock/easymock/pull/168#discussion_r52184125>:
>
> > @@ -373,6 +373,24 @@ public void asStub() {
> >          }
> >      }
> >
> > +    public IExpectationSetters<Object> atLeast(int minimalNumberOfTimes) {
> > +        try {
> > +            state.times(new Range(minimalNumberOfTimes, Integer.MAX_VALUE));
> > +            return this;
> > +        } catch (RuntimeExceptionWrapper e) {
> > +            throw (RuntimeException) e.getRuntimeException().fillInStackTrace();
> > +        }
> > +    }
> > +
> > +    public IExpectationSetters<Object> atMost(int maximalNumberOfTimes) {
> > +        try {
> > +            state.times(new Range(1, maximalNumberOfTimes));
>
> Ok, I understand.
> I remember that the Range class did not support a minimum lesser than 1.
> Maybe it is related to the fact mocks are typed by EasyMock.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/easymock/easymock/pull/168/files#r52184125>.
>